spec/controllers/state_file/questions/az_public_school_contributions_controller_spec.rb

```diff
@@ -82,36 +101,89 @@
       }.not_to change(intake.az322_contributions, :count)
     end
 
-    context "when 'no' was selected for made_contribution" do
-      before do
-        params[:az322_contribution][:made_contribution] = 'no'
+    context "with invalid params" do
+      render_views
+
+      context "yes/no question" do
+        it 'should not create the contribution when made_az322_contributions is missing' do
+          expect {
+            put :create, params: {
+              az322_contribution: {
+                state_file_az_intake_attributes: {
+                  made_az322_contributions: nil,
+                },
+                school_name: 'School A',
+                ctds_code: '123456789',
+                district_name: 'District A',
+                amount: 100,
+                date_of_contribution_month: '8',
+                date_of_contribution_day: "12",
+                date_of_contribution_year: Rails.configuration.statefile_current_tax_year
+              }
+            }
+          }.not_to change(Az321Contribution, :count)
+
+          expect(response).to render_template :new
+        end
       end
 
-      it "creates nothing" do
-        expect do
-          post :create, params: params
-        end.not_to change(Az322Contribution, :count)
+      context "contribution is invalid" do
+        let(:invalid_params) do
+          {
+            az322_contribution: {
+              state_file_az_intake_attributes: {
+                made_az322_contributions: "yes",
+              },
+              school_name: 'School A',
+              ctds_code: '123456789',
+              district_name: 'District A',
+              amount: 0, # must have amount
+              date_of_contribution_month: '8',
+              date_of_contribution_day: "12",
+              date_of_contribution_year: Rails.configuration.statefile_current_tax_year
+            }
+          }
+        end
+
+        it "renders new with validation errors" do
+          expect do
+            post :create, params: invalid_params
+          end.not_to change(Az322Contribution, :count)
+
+          expect(response).to render_template(:new)
+          expect(response.body).to include "must be greater than 0"
+        end
       end
     end
 
-    context "with invalid params" do
-      render_views
-
-      let(:invalid_params) do
+    context "DEPRECATED BEHAVIOR; keep tests until page is changed on prod" do
```

Comment:
When you make a ticket to remove the old page/behavior, can you include file & line numbers of what should be deleted?

Reply:
i haven't added those details yet but that's because i think it'll be a lot faster for me to just pick it up. will add all the info if someone else has to do it